src/mixins/helpers.js

```diff
@@ -206,12 +205,12 @@ var helpers = {
         swipeLeft: null
       };
 
-      callback = () => {
+      this.animationEndCallback = () => {
         this.setState(nextStateChanges);
         if (this.props.afterChange) {
           this.props.afterChange(currentSlide);
         }
-        ReactTransitionEvents.removeEndEventListener(this.refs.track.getDOMNode(), callback);
+        delete this.animationEndCallback;
```

Comment:
He didn't named `this.animationEndCallback = window.setTimeout` so there's no need of `clearTimeout` in this case.